### PR TITLE
fix draft discard by add workaround for , added more if statement to …

### PR DESCRIPTION
**Pull request by vinhtruong94**
…userResource

### CODE REVIEW

**Comment by vinhtruong94** [comment]
https://github.com/angular-ui/ui-router/issues/326
https://github.com/angular-ui/ui-router/issues/1901

File: public/app/app.js [hunk 1]
```diff
@@ -36,7 +36,7 @@
           && $stateParams.id) {
 
           event.preventDefault();
-
+          //this is broken
```
Comment (psaunders) [inline public/app/app.js]: If it's broken then you should fix it?

File: public/app/app.js [hunk 2]
```diff
@@ -46,17 +46,22 @@
                 .cancel('No thanks.');
 
           $mdDialog.show(confirm).then(function() {
+            // console.log($stateParams.id);
             motionResource.deleteMotion($stateParams.id);
+            console.log(toState);
```
Comment (psaunders) [inline public/app/app.js]: Console logging shouldn't be in production

File: public/app/app.js [hunk 3]
```diff
@@ -46,17 +46,22 @@
...
+            $state.go(toState.name, {'id': toParams.id} || toState.name || 'home');
           }, function() {});
 
         } else if (fromState.name === 'create-page' && $stateParams.id) {
           event.preventDefault();
 
           var confirm2 = $mdDialog.confirm()
             .parent(angular.element(document.body))
-            .title('Would you like to discard this draft?')
+            .title('Would you like to discard this fucking draft?')
```
Comment (psaunders) [inline public/app/app.js]: This isn't ready

File: public/app/app.js (hunk 1 above)
Comment (vinhtruong94) [inline public/app/app.js]: this is the initial comment I left while I first investigated the problem, I will remove the comment now.

File: public/app/app.js [hunk 4]
```diff
@@ -47,8 +46,12 @@
 
           $mdDialog.show(confirm).then(function() {
             motionResource.deleteMotion($stateParams.id);
+            console.log(toState);
```
Comment (psaunders) [inline public/app/app.js]: https://gist.github.com/guilherme/9604324

**Pull request merged**